src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,137 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        int year = 0;
+        int month = 0;
+        int dayOfMonth = 0;
+
+        int countForEach = 0;
+
+        for (String value : dateFrom.split("\\.")) {
+            if (countForEach == 0) {
+                dayOfMonth = Integer.parseInt(value);
+            } else if (countForEach == 1) {
+                month = Integer.parseInt(value);
+            } else if (countForEach == 2) {
+                year = Integer.parseInt(value);
+            }
+
+            countForEach++;
+        }
+
+        countForEach = 0;
+
+        LocalDate ldBegin = LocalDate.of(year, month, dayOfMonth);
+
+        year = 0;
+        month = 0;
+        dayOfMonth = 0;
+
+        for (String value : dateTo.split("\\.")) {
+            if (countForEach == 0) {
+                dayOfMonth = Integer.parseInt(value);
+            } else if (countForEach == 1) {
+                month = Integer.parseInt(value);
+            } else if (countForEach == 2) {
+                year = Integer.parseInt(value);
+            }
+
+            countForEach++;
+        }
+
+        countForEach = 0;
+
+        LocalDate ldEnd = LocalDate.of(year, month, dayOfMonth);
+        int countValuesInDataUser = 4;
+
+        String[] inputDataUser = new String[countValuesInDataUser];
+        Map<String, Integer> userData = new HashMap<String, Integer>();
+
+        for (int i = 0; i < data.length; i++) {
```

Comment:
давай зробимо рішення яке буде працювати для n-ої кількіості людей:  
```suggestion
         int salary = 0; // salary for each user
            for (String line : data) {
                arrayOfData = line.split(" ");
                LocalDate currentDate = LocalDate.parse(arrayOfData[0], FORMATTER);
                if (...compare names and data)
             //  increase salary here         
            }
            result.append(System.lineSeparator()).append(name).append(" - ").append(salary);
```

Reply:
Спасибо) Я конечно наворатил такого))) Не пойму как теперь узнать есть ли такое имя в массиве) Поищу ответ как сделать))